Tests/Functional/AbstractFunctionalTestCase.php

```diff
@@ -134,4 +136,17 @@ protected function setUpFrontendPage($pageId, array $typoScriptFiles = [], array
             GeneralUtility::writeFile($target, $fileContent);
         }
     }
+
+    /**
+     * Build and return instance of TcaDatabaseRecord based on TYPO3 version
+     */
+    protected function getTcaDatabaseRecordInstance(): TcaDatabaseRecord
```

Comment:
I don't think the introduction of a new function is necessary as obsolete constructor arguments are simply ignored.

Reply:
The constructor with required argument was introduced in TYPO3 13.4.3 with this change: https://github.com/TYPO3/typo3/commit/ba527f46e8859b346c46c2d3e69b991ebe77046c#diff-49fe48871c141f6093cc1c9406dcab02b788aa2e0e4e031d9f47d51a3f189a9bR27-R29

Reply:
I changed it and always pass the constructor argument directly